module/zfs/dmu_tx.c

```diff
@@ -884,6 +884,11 @@ dmu_tx_try_assign(dmu_tx_t *tx, uint64_t txg_how)
 		return (SET_ERROR(ERESTART));
 	}
 
+	if (!tx->tx_dirty_delayed &&
+	    dsl_pool_wrlog_over_max(tx->tx_pool)) {
```

Comment:
Let's add another dmu_tx kstat for this to count the number of times this happens.  Similar to `dmu_tx_dirty_delay` in the case below, how about `dmu_tx_wrlog_over_max`.

```c
DMU_TX_STAT_BUMP(dmu_tx_wrlog_over_max);
```

Reply:
Added. Here is the example.
```shell
root@dev:~# cat /proc/spl/kstat/zfs/dmu_tx
12 1 0x01 13 3536 104734659308608 105005405572681
name                            type data
dmu_tx_assigned                 4    302073
dmu_tx_delay                    4    0
dmu_tx_error                    4    0
dmu_tx_suspended                4    0
dmu_tx_group                    4    0
dmu_tx_memory_reserve           4    0
dmu_tx_memory_reclaim           4    0
dmu_tx_dirty_throttle           4    0
dmu_tx_dirty_delay              4    38840
dmu_tx_dirty_over_max           4    0
dmu_tx_wrlog_over_max           4    60
dmu_tx_dirty_frees_delay        4    0
dmu_tx_quota                    4    0
```